testsuite/openshift/objects/rate_limit.py

```diff
@@ -19,20 +20,43 @@ def create_instance(cls, openshift: OpenShiftClient, name, route: Referencable,
             "metadata": {"name": name, "namespace": openshift.project, "labels": labels},
             "spec": {
                 "targetRef": route.reference,
-                "rateLimits": [{"limits": []}],
+                "rateLimits": [],
             },
         }
 
         return cls(model, context=openshift.context)
 
+    @staticmethod
+    def _user_id_variable(user_id):
+        """Add configuration for rate limit."""
+        return {
+            "actions": [
+                {
+                    "metadata": {
+                        "descriptor_key": user_id,
+                        "default_value": "no-user",
+                        "metadata_key": {
+                            "key": "envoy.filters.http.ext_authz",
+                            "path": [{"segment": {"key": "ext_auth_data"}}, {"segment": {"key": user_id}}],
+                        },
+                    }
+                }
+            ]
+        }
+
     @modify
-    def add_limit(self, max_value, seconds, conditions: list[str] = None):
+    def add_limit(self, max_value, seconds, conditions: list[str] = None, variable=None):
```

Comment:
I would change variable for something more verbose, like `enable_auth`

Reply:
Do you want to use `enable_auth` like bool value here?

Reply:
I see that `variable` is element of `variables` list, but then it is "called" also `user_id` or `descriptor_key`. Moreover from the code I understand that "userID" is currently only possible value, I can't imagine current code would work with different value. So without deep knowledge I would say there should be better name and maybe for future "simple" value (str, bool) doesn't have to be enough.